core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -114,68 +125,69 @@ public TenantObject setEnabled(final boolean flag) {
 
     /**
      * Gets the subject DN of this tenant's configured trusted
-     * certificate authority.
+     * certificate authority. This method assumes that this tenant is
+     * configured with a single trusted CA.
      * 
+     * @deprecated replaced by {@link #getTrustedCaSubjectNames()}
      * @return The DN or {@code null} if no CA has been set.
      */
+    @Deprecated
     @JsonIgnore
     public X500Principal getTrustedCaSubjectDn() {
-
-        final JsonObject trustedCa = getProperty(TenantConstants.FIELD_PAYLOAD_TRUSTED_CA);
-        if (trustedCa == null) {
-            return null;
-        } else {
-            return Optional.ofNullable((String) getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_SUBJECT_DN))
-                    .map(dn -> new X500Principal(dn)).orElse(null);
-        }
+        final List<X500Principal> subjectdns = Optional.ofNullable(getTrustedCaSubjectNames()).orElse(null);
```

Comment:
IMHO we should remove this method altogether.

Reply:
i think if we should maintain the current contain model for backwards compatibility (as you mention previously), then we should keep this method too. Don't you think?

Reply:
this will probably be needed by the service implementation when adding/updating the trust anchor. However, the method should then return a `Set<X500Principal>`, don't you think?

Reply:
sure, but even if we were to return a list, it will not contain duplicate elements since the `trusted-ca` is internally implemented using a map with the `subject-dn` property serving as key.

will fix

Reply:
@Alfusainey I have added a [comment on the original issue](https://github.com/eclipse/hono/issues/623#issuecomment-496389453) which might shine some light on this ...

This basically goes back to whether we need to record a life span for the public keys or not.

Reply:
yes, am following the discussions there. however, I do not see a problem if the life span of the trusted-ca(s) are not registered with the tenant (but I might be wrong). 

Since a protocol adapter will retrieve a list of trusted CAs configured for a tenant, the adapter can try to validate a client's certificate using the first public key, if that fails, it tries the second public key, if that fails, it tries the third keyand so on, until it can finally find one that can valid a client's certificate.

This is because if one of the trusted CAs expires in the meantime, the tenant would have already  been configured with another (new) trusted CA which will succeed in authenticating the device.

Maybe you want protocol adapters be able to determine (upfront) which trusted CA can authenticate a client?